src/CharacterPositionController.test.js

```diff
@@ -0,0 +1,220 @@
+    };
+}
+
+function getCharacterPositionButton(characterPositionControllerWrapper, directionName) {
+    return characterPositionControllerWrapper.find('.CharacterPositionController__character-position-button').filter({value: directionName}).at(0);
+}
+
+function getCharacterPositionCoordinateBoxes(characterPositionControllerWrapper) {
+    return characterPositionControllerWrapper.find('.ProgramBlock__character-position-coordinate-box');
+}
+
+function getCharacterIcon(characterPositionControllerWrapper) {
+    return characterPositionControllerWrapper.find('.CharacterPositionController__character-column-character');
+}
+
+describe('Using change character position buttons', () => {
+    test.each([
+        'turnLeft', 'turnRight', 'up', 'right', 'down', 'left'
+    ])('Click/Press %s button ', (directionName) => {
+        expect.assertions(4);
+        const { wrapper, mockChangeCharacterPosition } = createMountCharacterPositionController();
+        const characterPositionButton = getCharacterPositionButton(wrapper, directionName);
+
+        characterPositionButton.simulate('click');
+        expect(mockChangeCharacterPosition.mock.calls.length).toBe(1);
+        expect(mockChangeCharacterPosition.mock.calls[0][0]).toBe(directionName);
+
+        characterPositionButton.simulate('keydown', { key: ' ' });
+        expect(mockChangeCharacterPosition.mock.calls.length).toBe(2);
+        expect(mockChangeCharacterPosition.mock.calls[1][0]).toBe(directionName);
+    });
+    test.each([
+        'turnLeft', 'turnRight', 'up', 'right', 'down', 'left'
+    ])('Click/Press %s button when editingDisabled Prop is true', (directionName) => {
+        expect.assertions(3);
+        const { wrapper, mockChangeCharacterPosition } = createMountCharacterPositionController({editingDisabled: true});
+        const characterPositionButton = getCharacterPositionButton(wrapper, directionName);
+        expect(characterPositionButton.get(0).props.className.includes('--disabled')).toBe(true);
+
+        characterPositionButton.simulate('click');
+        expect(mockChangeCharacterPosition.mock.calls.length).toBe(0);
+        characterPositionButton.simulate('keydown', { key: ' ' });
+        expect(mockChangeCharacterPosition.mock.calls.length).toBe(0);
+    });
+});
+
+describe('Using change character position by column/row labels', () => {
+    test('Changing x position', () => {
+        expect.assertions(6);
+        const { wrapper, mockChangeCharacterXPosition } = createShallowCharacterPositionController();
+        const characterXPositionCoordinateBox = getCharacterPositionCoordinateBoxes(wrapper).at(0);
+        const sampleXPosition = 'X';
+        const secondSampleXPosition = 'A';
+        const eventObject = (value) => (
+            {
+                key: 'Enter',
+                preventDefault: () =>  {},
+                currentTarget:
+                    { name: 'xPosition', value }
+            }
+        );
```

Comment:
I think this function is a good basis for making some general test utility functions. I'd suggest doing the following (and also removing the equivalent one in the `'Changing y position'` test):

- Split into 3 functions:
  - `makeChangeEvent()`
  - `makeBlurEvent()`
  - `makeKeyDownEvent()`
- Move them to a new `TestUtils.js` file so that they can be used in other tests

Reply:
Thanks for the comment @sbates-idrc, I see that making TestUtils would be helpful, but this case is specific to the CharacterPositionCoordinateBox, so it has specific values hard coded like key: 'Enter', and currentTarget object with name 'xPosition'. Are you thinking about specifying the hardcoded values as parameters to the functions?